*******************************
**** Handling Login Errors ****
*******************************
Currently login errors are generating a dev only error page and are not being displayed in any way to the user.
So to fix this I go back to the login form and wrap the 2 lines of "doSubmit" into a try/catch block.

I've been calling the errors "ex" so I'll continue that convention here.
I need to check if there is a response to ex and if that response has a status that === 400.
If that is true then display the error.
So first clone the errors object, then spread this.state.errors
then set errors.username = ex.response.data;
finally, setstate and pass the errors object.

final: 

doSubmit = async () => {
  try {
    const { data } = this.state;
    await login(data.username, data.password);
    } catch (ex) {
      if (ex.response && ex.response.status === 400) {
      const errors = {...this.state.errors};
      errors.username = ex.response.data;
      this.setState({ errors });
    }
  }
};

Next: Store the JSON web token on the client and redirect the user to the homepage.
